openedx/features/enterprise_support/api.py

```diff
@@ -233,52 +259,87 @@ def enterprise_enabled():
     return 'enterprise' in settings.INSTALLED_APPS and getattr(settings, 'ENABLE_ENTERPRISE_INTEGRATION', True)
 
 
-def enterprise_customer_for_request(request, tpa_hint=None):
+def enterprise_customer_for_request(request):
     """
     Check all the context clues of the request to determine if
     the request being made is tied to a particular EnterpriseCustomer.
     """
+
     if not enterprise_enabled():
         return None
 
     ec = None
+    sso_slug = request.GET.get('tpa_hint')
 
     running_pipeline = get_partial_pipeline(request)
     if running_pipeline:
         # Determine if the user is in the middle of a third-party auth pipeline,
-        # and set the tpa_hint parameter to match if so.
-        tpa_hint = Registry.get_from_pipeline(running_pipeline).provider_id
+        # and set the sso_slug parameter to match if so.
```

Comment:
@haikuginger Could you actually call this variable `provider_id`? That's how it's called in the third_party_auth app, that's obviously the attribute you're accessing here, and that's what it's called in the `enterprise_customer_identity_provider__provider_id` parameter below.

To me, `sso_slug` is too easily confused with [`idp_slug`](https://github.com/edx/edx-platform/blob/e8a36957b1f732974260e7b9b42b9c25148b492c/common/djangoapps/third_party_auth/models.py#L350) and [`provider_slug`](https://github.com/edx/edx-platform/blob/e8a36957b1f732974260e7b9b42b9c25148b492c/common/djangoapps/third_party_auth/models.py#L296) which are sometimes components of the `provider_id` but are not the same as the `provider_id`.